### ISSUE TITLE
Vulkan Issues / Discussion

### ISSUE THREAD

**Issue by DrChat** [benvanik/xenia#578]
~~**Note:** Add the flag --gpu=vulkan to enable the Vulkan backend!~~

**Comment by Gamedev1909** [comment]
[xenialogamd.txt](https://github.com/benvanik/xenia/files/299418/xenialogamd.txt)
ok here maybe you can find out why only amd crashes

**Comment by SakataGintokiYT** [comment]
1.Dragon Ball Z: Battle of Z- https://www.youtube.com/watch?v=lI7i18Q8DQ0
2.Fist of the North Star - https://www.youtube.com/watch?v=i_l1SwB-ebw

**Comment by Gamedev1909** [comment]
this long is for halo spartan assault AMDGpu 
[xeniaamdHSA.txt](https://github.com/benvanik/xenia/files/299420/xeniaamdHSA.txt)

**Comment by DrChat** [comment]
@Gamedev1909 Unfortunately the logs don't capture a stack trace - though they really should.
Can you attach a debugger and tell me where it crashes / generate a minidump and upload it somewhere?

**Comment by Gamedev1909** [comment]
which debugger do i need to attach ?

**Comment by DrChat** [comment]
@Gamedev1909  Visual Studio.

**Comment by Razzile** [comment]
I haven't got any logs or videos, but I've noticed that many games have inverted colours on the vulkan build. 

![image](https://cloud.githubusercontent.com/assets/5256381/15806979/7951ba4c-2b48-11e6-8a6c-89f31c302872.png)
opengl left, vulkan right
![image](https://cloud.githubusercontent.com/assets/5256381/15806997/45cd929e-2b49-11e6-9c46-3e3dca38fdeb.png)
vulkan left, opengl right

admittedly my OpenGL build of xenia is relatively old.

**Comment by Gamedev1909** [comment]
>	amdvlk64.dll!00007ff81393d3f4()	Unknown
 	amdvlk64.dll!00007ff81393a095()	Unknown
 	amdvlk64.dll!00007ff813938b34()	Unknown
 	amdvlk64.dll!00007ff813426948()	Unknown
 	amdvlk64.dll!00007ff813426e1c()	Unknown
 	amdvlk64.dll!00007ff81342022e()	Unknown
 	xenia.exe!00007ff744b705d9()	Unknown
 	xenia.exe!00007ff744b7033e()	Unknown
 	xenia.exe!00007ff744b671f9()	Unknown
 	xenia.exe!00007ff744b592c0()	Unknown
 	xenia.exe!00007ff744b58b8d()	Unknown
 	xenia.exe!00007ff744b591cf()	Unknown
 	xenia.exe!00007ff744b5899c()	Unknown
 	xenia.exe!00007ff744b58128()	Unknown
 	xenia.exe!00007ff744b5b3ad()	Unknown
 	xenia.exe!00007ff744b9567f()	Unknown
 	xenia.exe!00007ff744b93b30()	Unknown
 	xenia.exe!00007ff744a848fc()	Unknown
 	[External Code]	
this is from the stack after debugging it's pointing to this amdvlk64 having access violations

**Comment by DrChat** [comment]
Wonderful - it's crashing in the AMD drivers. Can you build Xenia from source so I can see where it is in Xenia?

**Comment by Gamedev1909** [comment]
ive never been able to do that but i can give you 100% control over my pc to do it if you want

**Comment by DrChat** [comment]
Where did you get your Xenia build from?

**Comment by Gamedev1909** [comment]
http://www.emucr.com/2016/06/xenia-vulkan-git-20160605.html right here

**Comment by Gamedev1909** [comment]
https://www.sendspace.com/file/6rvfa8 here is a video of the entire debug and here is the .sln i saved https://www.sendspace.com/file/14d1sr

**Comment by DrChat** [comment]
[Download this](https://dl.dropboxusercontent.com/u/5619434/Development/Xenia/Xenia.7z) and copy the stack trace here.

**Comment by Gamedev1909** [comment]
alright >	amdvlk64.dll!00007ffffc39d3f4()	Unknown
 	amdvlk64.dll!00007ffffc39a095()	Unknown
 	amdvlk64.dll!00007ffffc398b34()	Unknown
 	amdvlk64.dll!00007ffffbe86948()	Unknown
 	amdvlk64.dll!00007ffffbe86e1c()	Unknown
 	amdvlk64.dll!00007ffffbe8022e()	Unknown
 	xenia.exe!vkCreateGraphicsPipelines(VkDevice_T * device, VkPipelineCache_T * pipelineCache, unsigned int createInfoCount, const VkGraphicsPipelineCreateInfo * pCreateInfos, const VkAllocationCallbacks * pAllocator, VkPipeline_T * * pPipelines) Line 919	C
 	xenia.exe!xe::gpu::vulkan::PipelineCache::GetPipeline(const xe::gpu::vulkan::RenderState * render_state, unsigned __int64 hash_key) Line 264	C++
 	xenia.exe!xe::gpu::vulkan::PipelineCache::ConfigurePipeline(VkCommandBuffer_T * command_buffer, const xe::gpu::vulkan::RenderState * render_state, xe::gpu::vulkan::VulkanShader * vertex_shader, xe::gpu::vulkan::VulkanShader * pixel_shader, xe::gpu::PrimitiveType primitive_type, VkPipeline_T * * pipeline_out) Line 198	C++
 	xenia.exe!xe::gpu::vulkan::VulkanCommandProcessor::IssueDraw(xe::gpu::PrimitiveType primitive_type, unsigned int index_count, xe::gpu::CommandProcessor::IndexBufferInfo * index_buffer_info) Line 508	C++
 	xenia.exe!xe::gpu::CommandProcessor::ExecutePacketType3_DRAW_INDX_2(xe::RingBuffer * reader, unsigned int packet, unsigned int count) Line 1098	C++
 	xenia.exe!xe::gpu::CommandProcessor::ExecutePacketType3(xe::RingBuffer * reader, unsigned int packet) Line 585	C++
 	xenia.exe!xe::gpu::CommandProcessor::ExecutePacket(xe::RingBuffer * reader) Line 451	C++
 	xenia.exe!xe::gpu::CommandProcessor::ExecuteIndirectBuffer(unsigned int ptr, unsigned int count) Line 410	C++
 	xenia.exe!xe::gpu::CommandProcessor::ExecutePacketType3_INDIRECT_BUFFER(xe::RingBuffer * reader, unsigned int packet, unsigned int count) Line 746	C++
 	xenia.exe!xe::gpu::CommandProcessor::ExecutePacketType3(xe::RingBuffer * reader, unsigned int packet) Line 552	C++
 	xenia.exe!xe::gpu::CommandProcessor::ExecutePacket(xe::RingBuffer * reader) Line 451	C++
 	xenia.exe!xe::gpu::CommandProcessor::ExecutePrimaryBuffer(unsigned int read_index, unsigned int write_index) Line 388	C++
 	xenia.exe!xe::gpu::CommandProcessor::WorkerThreadMain() Line 156	C++
 	xenia.exe!xe::gpu::CommandProcessor::Initialize::__l2::<lambda>() Line 53	C++
 	[External Code]	
 	xenia.exe!xe::kernel::XHostThread::Execute() Line 1044	C++
 	xenia.exe!xe::kernel::XThread::Create::__l2::<lambda>() Line 381	C++
 	[External Code]	
 	xenia.exe!xe::threading::ThreadStartRoutine(void * parameter) Line 429	C++
 	[External Code]

**Comment by DrChat** [comment]
Okay that's not too good - it's crashing trying to compile a shader. I wonder if it's related to [this](https://devtalk.nvidia.com/default/topic/938071/vulkan/nvidia-shader-compiler-consumes-too-much-memory/).

**Comment by Gamedev1909** [comment]
i am working on compiling xenia on my rig now

**Comment by Gamedev1909** [comment]
ok wow i got a ton more info for you

**Comment by Gamedev1909** [comment]
https://www.sendspace.com/file/efu0k8 thats the build i compiled i got the same read oout as above no change

**Comment by DrChat** [comment]
Okay - try it again on the latest commit just to see if that fixed it.

**Comment by Gamedev1909** [comment]
ok one sec

**Comment by DrChat** [comment]
@Gamedev1909 Also - join #xenia-dev so we don't end up clouding this discussion.

**Comment by Gamedev1909** [comment]
is that on freenode?

**Comment by DrChat** [comment]
Yes.

**Comment by Gamedev1909** [comment]
i have to go out i will be back

**Comment by Gamedev1909** [comment]
Ok so what can be done I'm on mobile right now

**Comment by Gamedev1909** [comment]
ok im home i compiled the new build and still crashes >	http://pastebin.com/U0Lun7Js

**Comment by DrChat** [comment]
@Gamedev1909 Okay, make it crash again then go to Debug -> Save Dump As..., then save the minidump and upload it.

**Comment by Gamedev1909** [comment]
ok

**Comment by Gamedev1909** [comment]
i hope it helps

**Comment by DrChat** [comment]
Could you select "Minidump" in the save-as type instead of the one with the heap?

**Comment by DrChat** [comment]
Awesome - thanks!

**Comment by Gamedev1909** [comment]
anytime my amd gpu is at your command

**Comment by DrChat** [comment]
Can you regenerate the dump using the build I provided [here](https://dl.dropboxusercontent.com/u/5619434/Development/Xenia/Xenia.7z)?

**Comment by Gamedev1909** [comment]
yup let me boot into windows 10

**Comment by Gamedev1909** [comment]
doing it now

**Comment by Gamedev1909** [comment]
https://www.sendspace.com/file/0oag1y here ya go

**Comment by SonofUgly** [comment]
The red/blue issue looks the same as xenia-project/game-compatibility/issues/206 and #460

**Comment by Gamedev1909** [comment]
F> 00000024 Device::ResolvePath()
G> 00000004 Generated pixel shader (36b) - hash 52910627A8EDB5CE:
/*    0.0 */       alloc colors
/*    0.1 */       exece 
/*    1   */          max oC0, r0, r0
that last build got a bit farther i checked the last few logs and this is new

**Comment by Jonatino** [comment]
@DrChat Crashing with Jun 6 build (attempting to load Red Dead Redemption).

Screenshot: https://dl.dropboxusercontent.com/u/91292881/ShareX/2016/06/2016-06-06_19-35-09.png

*Currently upload dmp file*

**Comment by sephiroth99** [comment]
I also have been trying Vulkan on AMD. 

Tried today with the latest everything. (Xenia-vulkan commit b01903cc and Crimson 16.6.1)

Some quick observations (that you may already know):
- vkCreateGraphicsPipelines fails on the first translated shader pipeline. The two pipelines of the immediate drawer are created successfully.
- re: memory usage, no weird memory usage seen in the task manager.
- I tried disabling the pipeline cache, no change.
- I'm pretty sure no Vulkan debug layer will help here, it really looks like AMD's SPIR-V compiler crashing hard.
- Speaking of the crash, the assembly of where it crashes is ```cmp         byte ptr [rax+20h],sil```, where rax is NULL. When the 2 immediate drawer pipelines are created, this value is obviously not NULL. rax comes from another memory area that really looks like an internal data structure of the driver...

Anyway, I'm not that surprised Vulkan crashes. The GLSL version of the translated shaders on AMD still have issues (see #416 for example). As the generated SPIR-V shaders have a very similar approach as the generated GLSL shaders (I assume, unless I'm wrong?), my guess is that it is the same bug, but manifesting differently because of the different backend.

**Comment by DrChat** [comment]
@sephiroth99 If you can figure this out - that would be wonderful. I can't install AMD drivers on this computer which cripples my debugging capabilities.

Set a memory breakpoint on that internal data structure and see if you can find where the drivers touch it. May not be too straightforward unfortunately if it's allocated in different regions of memory every time Xenia is restarted.
If it's touched by another API call - wonderful! If not, we might need to submit a dump to AMD and get their help.

**Comment by Gamedev1909** [comment]
Download GPuOpen there might be a few things you could use or download the mesa drivers for linux they are open source and have vulkan in them as well those guys could help also

**Comment by Gamedev1909** [comment]
https://github.com/GPUOpen-LibrariesAndSDKs/HelloVulkan

**Comment by Parovozik** [comment]
Hey. I tried Bomberman Live! and results:
http://imagehost.spark-media.ru/i4/11B2A8AA-3016-8E0F-B4D9-FE95808E4713.jpg Original Xbox 360 Picture
http://imagehost.spark-media.ru/i4/11B2A8AA-3016-8E0F-B4D9-FE95808E4713.jpg OpenGL Picture
http://imagehost.spark-media.ru/i4/BEC1F320-3435-B932-31D3-29E5E2802584.jpg Vulkan api Picture
Xenia builds from June 6, 2016
p.s. 'Bomberman Battlefest' on Vulkan make crash after load game level (OpenGL work equally Bomberman Live!)

**Comment by DrChat** [comment]
@sephiroth99 / @Gamedev1909 Can you generate a crash dump from Banjo Kazooie/Tooie and upload it with amdvlk64.dll?
I'm gonna try and collect some information to post a thread on AMD's forums.

**Comment by Gamedev1909** [comment]
https://www.sendspace.com/file/u4ahyz you need the crash dumb from the other guy as i dont have that game but here is the amdvlk64

**Comment by DrChat** [comment]
@Gamedev1909 The file you uploaded appears to be corrupted - could you try it again?

**Comment by Gamedev1909** [comment]
[amdvlk64.dll.zip](https://github.com/benvanik/xenia/files/305707/amdvlk64.dll.zip)
compressed on os x so please use winrar to unzip

**Comment by DrChat** [comment]
@Gamedev1909 Thanks - but looks like your drivers have been updated since you created that last dump. Can you recreate the dump and upload it again?

**Comment by Nmzik** [comment]
@DrChat Mafia 2 demo goes in-game in latest "master build". In "vulkan build" it's just black screen (no intro + no gameplay but it still goes in-game)
![pic](https://cloud.githubusercontent.com/assets/691945/15912931/faf5e962-2ddf-11e6-9317-f474941bba20.PNG)

**Comment by sephiroth99** [comment]
I tried to see from where the NULL value came from, but it does not seem straight forward, at that point it's practically reverse engineering AMD's driver... 

I tried the trick I try with the OpenGL backend to disable the geometry shader (on OpenGL AMD it shows something instead of a black screen), but it did not change anything on the Vulkan side (still crashes).

I don't have Banjo games, I do my tests with Forza 2. If you have it I can send you a minidump and my vulkan dll.

**Comment by DrChat** [comment]
@sephiroth99: I've got forza motorsport 2 - Go ahead and send me a minidump and amdvlk64.dll

**Comment by krzysztof860** [comment]
@DrChat 
I generated a crash dump from Banjo Tooie. http://www88.zippyshare.com/v/e1oNUIE2/file.html
Also I have a amdvlk64.dll - http://www63.zippyshare.com/v/2QLUcdfc/file.html
Drivers are newest (16.6.1). My gpu is R9 290.

**Comment by DrChat** [comment]
@krzysztof860 Thanks! But I also need xenia.exe and xenia.pdb too since you built it yourself :P

**Comment by krzysztof860** [comment]
@DrChat 
Here you go: http://www69.zippyshare.com/v/Yk64As64/file.html (exe and pdb)
I also tried Banjo Kazooie. Dump: http://www52.zippyshare.com/v/wEwWOePP/file.html

**Comment by DrChat** [comment]
Excellent - thanks.

**Comment by DrChat** [comment]
@krzysztof860 If you can - do me a favor: set a breakpoint in amdvlk64.dll where it crashes and restart Xenia so it hits the breakpoint before it crashes.

**Comment by DrChat** [comment]
Okay AMD people - try it again with 7ef7cc39f586f4c06f0801955bcbac243f32b46a.

**Comment by Gamedev1909** [comment]
can you make a build of that for me this is a new insider build and i dont have time to install any tools

**Comment by DrChat** [comment]
[Here you go](https://dl.dropboxusercontent.com/u/5619434/Development/Xenia/Xenia.7z)

**Comment by Gamedev1909** [comment]
its fighting to run games taking so long to load

**Comment by woody2014** [comment]
making my cpu freeze

**Comment by Gamedev1909** [comment]
[XeniaDump.zip](https://github.com/benvanik/xenia/files/310860/XeniaDump.zip)

**Comment by DrChat** [comment]
What games are you trying? For now you should keep it limited to games that are proven to work already.

**Comment by Gamedev1909** [comment]
red dead im about to try halo wars and minecraft and halo 4 all known to work

**Comment by Gamedev1909** [comment]
trying to run a game again says i can't use vulkan loool

**Comment by DrChat** [comment]
Let's bring this discussion to irc.freenode.net #xenia-dev for now.

**Comment by nulcearc** [comment]
vulkan no difference on my gpu. it crashes on vkCreateGraphicsPipeline
GL4.5 shows black screen with sound. gpu amd radeon 8750m game Sonic CD

**Comment by krzysztof860** [comment]
@DrChat 
I don't know if I did it properly (Debug>New Breakpoint>Function Breakpoint and I entered pVertexInputState in Function Name).
There is dump from Banjo Tooie: http://www107.zippyshare.com/v/DQRaCUQN/file.html
Exe and pdb: http://www18.zippyshare.com/v/G2gLHKwl/file.html

**Comment by SakataGintokiYT** [comment]
@Gamedev1909 
@nulcearc 
@krzysztof860 

Try --> http://www22.zippyshare.com/v/6NNkYVt2/file.html

**Comment by Gamedev1909** [comment]
thats a no go but thanks seems xenia busted vulkan on my system

**Comment by Razzile** [comment]
Whilst probably not related directly to vulkan, I've noticed in the latest code audio is quite a lot slower than before. This is odd as there hasn't been any commits to /apu for 6 months.

**Comment by DrChat** [comment]
@razzile Could you clarify which games you're seeing this behavior on? Does this slowdown also happen on the GL backend?

**Comment by Razzile** [comment]
Forza motorsport 2 and 3D Ultra Minigolf 2 are the games I tested, both didn't have audio slowdowns previously. let me just test opengl quick

**Comment by Razzile** [comment]
Okay I rebuilt xenia using the exact same commit and it's suddenly full speed again. Was likely a problem my end

**Comment by Gamedev1909** [comment]
hey if i make a thread about getting a game working would you help me collect everything i need to help you code for it

**Comment by ScorchEmber256** [comment]
This isn't directly related to the vulkan builds, but has anyone noticed that TLoS:DOTD crashes the videocard? No one has said anything about it, and its happened since late November builds on the GL backend. It works in vulkan for some reason but not in the master builds.
[xenia.txt](https://github.com/benvanik/xenia/files/322010/xenia.txt)

**Comment by ScorchEmber256** [comment]
Upon investigation I found the specific commits that messed up numerous things in almost every gameplay accessible game. 

From commit #518 fixed some shader issues in spyro which was amazing however commit #514 caused the viewport to go out of whack before crashing within seconds. But it didn't crash the graphics card. After #520 it started crashing the graphics card with no visuals ingame. So I think maybe reviewing those commits would be beneficial for a number of games. Sorry if this is a bit indecisive, isolatig commits on Appveyor is annoying XD.

Prevent some divide by zeros in shaders :
![xenia detached_2e5c64f_jan 18 2016 - legend of spyro 19_06_2016 9_19_28 am](https://cloud.githubusercontent.com/assets/13728013/16174346/eb834ef6-3600-11e6-8b0b-45802b59b46c.png)
  
x64 Speedups (I think #524 may have been commited as well):
![xenia detached_0e11111_jan 14 2016 - legend of spyro 19_06_2016 9_18_39 am](https://cloud.githubusercontent.com/assets/13728013/16174356/8d519828-3601-11e6-9958-6fec74bab4ec.png)

**Comment by JohnGodgames** [comment]
@ScorchEmber256 I've mentioned broken TLoS:DOTD a long time ago, but only on IRC chat I think and then it was propably forgotten.Benvanik brokes it when he was trying to fix https://github.com/benvanik/xenia/issues/473 propably.

**Comment by DrChat** [comment]
[Posted a thread](https://community.amd.com/message/2731032) on the AMD forums. Hopefully an AMD dev will lead me in the right direction.

**Comment by Gamedev1909** [comment]
if you need to email roy he will get you in direct contact with driver devs also i suggest signing up to get on IRC channel #radeon on irc.freenode.net for radeon users and developers.

**Comment by Gamedev1909** [comment]
Drchat you got a forum reply from a amd dev

**Comment by raven02** [comment]
It would be nice if can have a simply option to switch the GL and Vulkan backend

**Comment by jackchentwkh** [comment]
raven02 <notifications@github.com> 於 2016年6月23日 星期四寫道：

> It would be nice if can have a simply option to switch the GL and Vulkan
> backend
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/benvanik/xenia/issues/578#issuecomment-228000755>, or mute
> the thread
> <https://github.com/notifications/unsubscribe/APnC7QR4h1zkVfh68ja3SaNXbMYRBWuzks5qOlVHgaJpZM4IuZSM>
> .
>


-- 
thanks and regards!

Jack Chen
Substrate sales and tech. service
YXsemicon Ltd.
Cell: +886-933311773
Office:+886-7-7536832
Fax:+886-7-7431976

**Comment by JohnGodgames** [comment]
I will post some screenshots to current comparison ogl vs. vulkan backend.

**Air Conflicts: Secret Wars**
![air secret ogl vs vulkan](https://cloud.githubusercontent.com/assets/11453507/16305199/1ad155c6-3959-11e6-8fb5-ae6abc46b268.jpg)

**Iron Man**
![iron man ogl vs vulkan](https://cloud.githubusercontent.com/assets/11453507/16305013/4a4d00b2-3958-11e6-9a37-4d83a378ae2c.jpg)

**Virtua Tennis 3**
![virtua tennis 3 ogl vs vulkan](https://cloud.githubusercontent.com/assets/11453507/16304555/35c13f70-3956-11e6-8740-4bfc2932bad9.jpg)

**Comment by gibbed** [comment]
The darker colors in some games is likely due to swapped channels, so it could probably be considered all one issue.

**Comment by ikki84** [comment]
How about a DX12 backend to windows 10 users?

**Comment by gibbed** [comment]
With GL being deprecated I don't think there are any plans to work on new GPU backends. Developing Vulkan is probably the priority currently.

**Comment by DrChat** [comment]
As I've said before - no existing contributors plan on creating a DX backend. You are completely free to do so if you want to.

**Comment by Margen67** [comment]
Is Vulkan eventually going to replace OGL?

**Comment by theoldsport** [comment]
@Margen67 Eventually, yes

**Comment by DrChat** [comment]
AMD people: Try 036df2ce126c1ab8d1578d627c4b3e5ea4969be0 - it might be fixed now.

**Comment by raven02** [comment]
@DrChat , just wonder how to switch to Vulkan backend?

**Comment by DrChat** [comment]
@raven02 See OP.

**Comment by woody2014** [comment]
where do we add --gpu=vulkan

**Comment by woody2014** [comment]
still crashes

**Comment by krzysztof860** [comment]
@DrChat Unfortunately it still crashing. Tested on Banjo Tooie.
Minidump - http://www112.zippyshare.com/v/Pl2bTsD8/file.html
Exe and pdb - http://www48.zippyshare.com/v/JFmAQ7SC/file.html

**Comment by DrChat** [comment]
@krzysztof860 - Can you attach the log file as well?

**Comment by krzysztof860** [comment]
@DrChat Here you go - http://www94.zippyshare.com/v/jVNNgujm/file.html

**Comment by DrChat** [comment]
Neat - this looks like a new crash now.

**Comment by raven02** [comment]
thanks @DrChat

**Comment by DrChat** [comment]
Okay - AMD is gonna be on hold til Monday.

**Comment by SakataGintokiYT** [comment]
Dragon Ball Z Budokai HD Collection:
<img src="http://www.fotoszok.pl/upload/eddb8b64.jpg" alt="Fotoszok.pl" />
<img src="http://www.fotoszok.pl/upload/75bd4164.png" alt="Fotoszok.pl" />
___________________________________________________________________

<img src="http://www.fotoszok.pl/upload/eaf823dd.jpg" alt="Fotoszok.pl" />
<img src="http://www.fotoszok.pl/upload/b70c2ad8.jpg" alt="Fotoszok.pl" />

**Comment by Nukem9** [comment]
I think this crash was posted before, but I can't find it right now.

http://i.imgur.com/06LFd9M.png
COD4 SP @ first mission - crashes in nvidia's nvoglv64.dll

**Comment by SakataGintokiYT** [comment]
*Naruto: Rise of a Ninja
Unhandled execption at 0x00000000023251fc (nvoglv64.dll)

Output : https://gist.github.com/SakataGintokiYT/d43af27cfdb52dae185e247d678673ca

**Comment by DrChat** [comment]
AMD people - Try it again with 75908a9865b27b2eeaa21a15e3b14c9777b00d56.

**Comment by sephiroth99** [comment]
Tried 9abfc52 still crashes at the same place.

I had tried the SPIR-V validator before, it did not find any issues; it is still very much a work in progress.

I will upload the newly generated shaders to the AMD thread.

**Comment by DrChat** [comment]
@sephiroth99 Alrighty - if you want you can take over that thread and I'll keep an eye on it.

**Comment by DrChat** [comment]
Neat - AMD is no longer crashing as of 14a6fd838a362031195de05a37fafd0941e689a9.

**Comment by Gh0stblade** [comment]
I'm still getting crashes.

**Comment by DrChat** [comment]
@Gh0stblade Which games?

**Comment by SakataGintokiYT** [comment]
@krzysztof860 
@woody2014 
@Gamedev1909 

Try latest build

**Comment by woody2014** [comment]
Good job it`s not crashing But have a blank screen.

**Comment by Gh0stblade** [comment]
@DrChat Nevermind, no crashes. Looks like Github desktop app didn't properly pull down the latest commits.

@woody2014 Same here, just black screen.

**Comment by Gamedev1909** [comment]
could someone link the newest build please thanks

**Comment by calvin--** [comment]
@DrChat - Can confirm that AMD doesn't crash anymore, but that it only shows black screen. Tested Banjo and Kazooie and Bomberman. They both play audio and accept input, just no video output.

**Comment by DrChat** [comment]
Wonderful - Someone with experience in renderdoc, can you capture a trace and start analyzing it?
Launch Xenia with renderdoc using these flags: "--flagfile=scratch/flags.txt --vulkan_primary_queue_only=true"

When in-game, press F4 to capture a trace and close Xenia after it's finished writing it.

I can't do anything with the tracefile from here, so don't upload it.

**Comment by Gamedev1909** [comment]
i tried adding those flags but it doesnt open

**Comment by woody2014** [comment]
can we open a file in the vulkan backend then trace file.

**Comment by calvin--** [comment]
@Gamedev1909 - did you build it yourself? Otherwise you wont have the flags.txt file and it wont open. I managed to generate the trace, but renderdoc can't open the file, so I'm probably doing the wrong. I'm not really qualified to debug it anyway, but if you have any insight into why it wont open (says wrong magic header) @DrChat ?

**Comment by Gamedev1909** [comment]
yes i built it my self i do get a -- jit error when trying to run some games

**Comment by Gamedev1909** [comment]
has this black screen vulkan issue been fixed and no one said anything ?

**Comment by DrChat** [comment]
Nope. Still need someone competent to run a trace on an AMD card.

**Comment by Gamedev1909** [comment]
i can't get your trace flags to work but i did get a trace for you 
[Trace.zip](https://github.com/benvanik/xenia/files/360838/Trace.zip)

**Comment by ikki84** [comment]
Tested with 16.7.2 Hotfix, still the same.
Radeon R7 200

**Comment by Gamedev1909** [comment]
DrChat since those flags you said to use casues a crash if there any other thing i can do to gather a trace? besides what ive done insider builds are over for now

**Comment by RenegadeSaiyan** [comment]
So now all Xenia does with Vulkan is Crash

**Comment by ikki84** [comment]
Tested again with Minecraft.
Crashed after load the file.

-Radeon R7 200
-Driver 16.7.3 AMD Crimson Edition 28/07/2016
-Xenia build f81a99e 29/07/2016

**Comment by RenegadeSaiyan** [comment]
I tried some games that were proven to work, same result.

**Comment by RenegadeSaiyan** [comment]
I tried Sonic 2006, And Naruto Storm 2

**Comment by DrChat** [comment]
AMD people: I'm working on the crashes and blackscreens - be patient.

**Comment by Gamedev1909** [comment]
i am about to help you with crash reports and a trace if i can

**Comment by Gamedev1909** [comment]
[xeniaDump.zip](https://github.com/benvanik/xenia/files/389850/xeniaDump.zip)
newest 16.7.3 drivers on windows 10 RS1 14393.5 RTM

**Comment by Gamedev1909** [comment]
also i am learning Vulkan to try and help in the future

**Comment by SakataGintokiYT** [comment]
**Progress  :)** 
<img src="http://www.fotoszok.pl/upload/4715bfa0.jpg" alt="Fotoszok.pl" />

**Comment by Gamedev1909** [comment]
can you link me that build ? and was it on amd?

**Comment by SakataGintokiYT** [comment]
@Gamedev1909 
Build from appveyor 
https://ci.appveyor.com/project/benvanik/xenia/build/1.0.451-master/artifacts

**Comment by Gh0stBlade** [comment]
@SakataGintokiYT 

I get some exception here (AMD): 
xenia.exe!xe::gpu::vulkan::CachedRenderPass::CachedRenderPass(VkDevice_T * device, const xe::gpu::vulkan::RenderConfiguration & desired_config) Line 441	C++

**Comment by Gamedev1909** [comment]
solid build for opengl 4.5 but vulkan i hear sonic 06 and bam crash

**Comment by Gamedev1909** [comment]
well now opengl is broken and vulkan

**Comment by Margen67** [comment]
Vulkan backend in a nutshell: https://youtu.be/68ugkg9RePc

**Comment by SakataGintokiYT** [comment]
@DrChat 
look at this :  [Work around semi-broken dual-source blending on AMD.]  / Dolphin 
https://github.com/dolphin-emu/dolphin/pull/3935/commits/e253a1673905b0b7df1fc59035b416c53b6d8d85

**Comment by sephiroth99** [comment]
I think the current issue with Vulkan crash on AMD is due to this (extracted from [Vulkan spec here](https://www.khronos.org/registry/vulkan/specs/1.0/xhtml/vkspec.html#VkAttachmentDescription)):

> Multiple attachments that alias the same memory must not be used in a single subpass

I tried to separate the 4 color attachments into 4 subpasses, and the subpass create command does not crash anymore. Now, of course the rest of the code is not made for 4 subpasses so I had other issues e.g. dependencies, but the crash is due to the single subpass I think.

**Comment by krzysztof860** [comment]
Sorry for off-topic, but @DrChat someone wants to give you a R9 270X (2GB)
https://www.reddit.com/r/xenia/comments/4y25ag/help_with_donating_a_gpu_to_the_xenia_devs/

**Comment by paulsapps** [comment]
I would assume he wants a full system, ripping out a working gpu in your own personal system with one some randomer has sent you to fix 1 issue probably isn't a great idea :s

**Comment by Gamedev1909** [comment]
this isnt one issue this is a huge issue for everyone with amd cards so it's a good thing

**Comment by SakataGintokiYT** [comment]
@DrChat 
**Progress !!!**
https://github.com/benvanik/xenia/commit/9457c63a645df723b422f73a704cb6adf8b779c1
**Xenia Vulkan build : [9457c63]**
***Video : https://www.youtube.com/watch?v=GUzDHodfmE4***
<img src="http://www.fotoszok.pl/upload/c6499918.jpg" alt="Fotoszok.pl" />

**Comment by Gamedev1909** [comment]
Vulkan proves to be a giant turd for amd i hope someone adds direct x 12 soon that api is hard to mess up on any gpu

**Comment by Margen67** [comment]
I hope DX12 is added anyway so the speed can be compared to Vulkan.

**Comment by Deathlock47** [comment]
All NFS games using EAGL4 (Most Wanted, Carbon and Prostreet) crash at startup on Vulkan despite being close to playable on OpenGL
Here's the [Carbon log](https://gist.github.com/Deathlock47/10fb119e9752b6cc66c2d03fd3e560b1) for example

**Comment by ikki84** [comment]
any news?

**Comment by nuclearc** [comment]
amd released new drivers that 'support' glsl 4.5

**Comment by Razzile** [comment]
only a year or so too late lmao

**Comment by Gamedev1909** [comment]
its never to late

**Comment by ikki84** [comment]
I will update the driver and test here.

**Comment by Margen67** [comment]
@Gamedev1909 https://youtu.be/lL2ZwXj1tXM

**Comment by Gamedev1909** [comment]
@DrChat hey tried the newest build still a no go do you need anything a dump maybe?

**Comment by Gamedev1909** [comment]
[xenia.zip](https://github.com/benvanik/xenia/files/588349/xenia.zip)
here is a dump for kicks

**Comment by NuclearC** [comment]
anyone thinks to write D3D12 back-end for Xenia?

**Comment by Gamedev1909** [comment]
That would have been the best idea from the start not sure why it wasnt done. Vulkan was a mistake sadly as its way to busted

**Comment by gibbed** [comment]
The Vulkan backend is a work in progress.

**Comment by NuclearC** [comment]
but fcking DirectX 11 is faster than Vulkan

**Comment by gibbed** [comment]
If you think you can do better, feel free to implement a DX11 or DX12 backend yourself.

**Comment by NuclearC** [comment]
i didnt say i can do it better, but i want to understand why @benvanik removed DX11 backend from start

**Comment by gibbed** [comment]
Because DX11 was not meeting the demands of the emulator.

**Comment by NuclearC** [comment]
btw im getting 'Access violation reading location' exception on this line when using Vulkan on AMD driver:
```C++
return disp->CreateRenderPass(device, pCreateInfo, pAllocator, pRenderPass);
```
Thats the line 1653 in file 'trampoline.c' function vkCreateRenderPass which is called from CachedRenderPass(...)
Here is the image:
![image](https://cloud.githubusercontent.com/assets/8341270/21955657/013e6c58-da89-11e6-8242-9055ec8906b1.png)

What can i do to fix this?

**Comment by ikki84** [comment]
@NuclearC Increasing the number of good contributors. Capable people that want to spend free time solving this issue. 

I remember very well in 2015 when someone asked about the issues about AMD videocard, and the answer was something like "wait for Vulkan to be released". 
After Vulkan, the answer was "AMD should fix the driver", and now they say "You are free to do it yourself".

That doesn´t mean that they are not working hard but maybe people from another emulator team could help.

**Comment by gibbed** [comment]
@ikki84 The solution for Vulkan issues is not "make DX11/DX12 backend instead", which is what I was talking about.

For @NuclearC's issue, the stack trace clearly shows a crash in AMD's code, which is extremely hard to debug without having access to AMD hardware (at least for myself, I do not have any AMD hardware).

I'd be happy to see more contributors, but I have no interest in soliciting other projects for them. Genuine interest is key.

**Comment by Gh0stBlade** [comment]
I do have AMD hardware and also experience the same issues.

**Comment by rlabrecque** [comment]
You should send your AMD hardware to a member of the team (most likely DrChat) so that someone can actually fix it then.

**Comment by Gamedev1909** [comment]
I would be happy to lend you my desktop remotely to debug

**Comment by DrChat** [comment]
@Gamedev1909 Okay - hop on IRC @ #xenia-dev so we can discuss doing that.

**Comment by Gamedev1909** [comment]
I can on monday ? If my daughter isnt born that day my email is gamedev1909@gmail.com

**Comment by neolightning** [comment]
i would be willing to help as i have AMD, just depends if i have the time.

**Comment by DrChat** [comment]
@Gamedev1909 Congrats, good luck to you. Whenever you're free just hop on. Same to you @neolightning

**Comment by Gamedev1909** [comment]
https://drive.google.com/open?id=0B4wd1D7L6SRYVlJnLS0xTWFnYkU here is the codeXL debug info the only mention i saw of shaders was in Index, CmdBuffer #, Call, Parameters, GPU Time, Result, Start Time, End Time
11, CmdBuffer3, vkCmdPushConstants, 0x00000216cfc94f90,0x00000216cfcbe9c0,VK_SHADER_STAGE_VERTEX_BIT,0,64,0x00000083366ff0d0, after that i saw nothing relating to shaders besides GCN shaders

**Comment by NuclearC** [comment]
I have AMD too, i want to help :D xD

**Comment by Gamedev1909** [comment]
Hey drchat im at the hospital my wife is having contractions and my pc is booted into macos 10.12.2 so use nuc  if you want

**Comment by haneefmubarak** [comment]
@Gamedev1909 slightly TMI on that last note but congratulations! Have you two decided what you'll name her?

**Comment by Gamedev1909** [comment]
Olivia like olivia benson from law and order svu

**Comment by woody2014** [comment]
Vulkan no longer crashes with latest drivers Crimson ReLive Edition 17.1.1 but shows black screen.

**Comment by Gamedev1909** [comment]
Yup i let drchat debug on my system

**Comment by woody2014** [comment]
Are we finally gonna get and Vulkan working

**Comment by Gamedev1909** [comment]
Maybe after more work

**Comment by NuclearC** [comment]
Again crashing in debug mode:
![image](https://cloud.githubusercontent.com/assets/8341270/22402627/621533f6-e617-11e6-9baa-74432f9f8d4c.png)

Maybe this makes black screen?

**Comment by 64fanatic** [comment]
Tried out Deadly Premonition (Red Seeds Profile) on Xenia. I had tested this a year or two ago and the game never got past the menu, Vulkan gets ingame with graphics at full speed now which is neat! Still hilariously broken with the text jumbled with English and Russian? I think. I didn't think York could look any more creepy than he already did.

**Vulkan:** 
https://youtu.be/hLiI6pgbXHs

**OpenGL:**
https://youtu.be/IfMNkyaiUDY

**Comment by Margen67** [comment]
@DrChat --gpu=vulkan isn't needed anymore.

**Comment by XeClutch** [comment]
@Margen67 that's because Vulkan has become the default renderer for Xenia.

**Comment by Margen67** [comment]
Exactly so the issue should be updated.

**Comment by Gamedev1909** [comment]
in IRC is there any reason it wont let me talk in Xenia-dev channel ? says im banned from talking but i only joined to help you out dr chat?

**Comment by Margen67** [comment]
You need permission to talk.

**Comment by maxton** [comment]
You only need to be identified with NickServ afaik

**Comment by Gamedev1909** [comment]
hey DrChat i think it's time we do more remote debugging set a day and time let's get vulkan working for amd

**Comment by NuclearC** [comment]
i cant wait for vulkan version to work with AMD hardware :D

**Comment by Gamedev1909** [comment]
newest build crashes amd again

**Comment by ikki84** [comment]
@Gamedev1909 MK vs DC not crashing here, just black screen as usual.

**Comment by DrChat** [comment]
Hit F3, can you see the profiler?

**Comment by ikki84** [comment]
Yes

**Comment by DrChat** [comment]
What games are you trying on AMD?

Can you try banjokazooie / banjotooie / supermeatboy / marbleblast / gripshift?

**Comment by ikki84** [comment]
@DrChat Tested supermeatboy, also shows sound, not shows any rendering on screen. 
Also I had to try to load it 4 times, crashed randomly in 3 tries...

**Comment by Gamedev1909** [comment]
According to my friend slash sonic unleashed renders images and works a bit https://cdn.discordapp.com/attachments/223866782539186177/280858668692668426/2017-02-13_16-31-48.png

**Comment by DrChat** [comment]
Where does AMD crash in super meat boy? Can you give me a callstack?

**Comment by Gamedev1909** [comment]
So after testing minecraft and hitting f3 i see 
![capture](https://cloud.githubusercontent.com/assets/18043258/22947682/66fb4262-f2d2-11e6-89c6-31fb98db3ef3.JPG)

**Comment by DrChat** [comment]
I've spotted a point in the render cache where our API usage is invalid @ [render_cache.cc#425](https://github.com/benvanik/xenia/blob/6bbd399cfd947eae3872bf7091ffe2941cf1b884/src/xenia/gpu/vulkan/render_cache.cc#L425)

According to API specification: "Multiple attachments that alias the same memory **must not** be used in a single subpass." We alias attachments which aren't written to.

AMD people: As a hack, change colorAttachmentCount to 1 on line 430. In [pipeline_cache.cc#1410](https://github.com/benvanik/xenia/blob/6bbd399cfd947eae3872bf7091ffe2941cf1b884/src/xenia/gpu/vulkan/pipeline_cache.cc#L1410), echo the same change.

**Comment by Gamedev1909** [comment]
https://www.youtube.com/watch?v=DaojxkQ1Y8Q drchat you are going to love this

**Comment by DrChat** [comment]
Awesome - can I get some screenshots from one of the games I mentioned earlier?

**Comment by ikki84** [comment]
Changed the src/xenia/gpu/vulkan/pipeline_cache.cc 
-  state_info.attachmentCount = 4;
+  state_info.attachmentCount = 1;

also in  src/xenia/gpu/vulkan/render_cache.cc 
-  subpass_info.colorAttachmentCount = 4;
+  subpass_info.colorAttachmentCount = 1;

now, SMB always crash for a reason that I don´t know. Here is the log.
[smb.zip](https://github.com/benvanik/xenia/files/775555/smb.zip)

**Comment by neolightning** [comment]
Let me know if there is anything to test, i do have Marble Blast on my pc if needed for that specific game

**Comment by ikki84** [comment]
@DrChat here is another try log.
[xenia.zip](https://github.com/benvanik/xenia/files/775560/xenia.zip)

**Comment by DrChat** [comment]
@ikki84 The logs are useless in this case. Can you provide me with a callstack of the crash?

**Comment by Gamedev1909** [comment]
i dont have those games but i will buy em soon can someone post that hacked build

**Comment by neolightning** [comment]
I can send Marble Blast if you give me a few

**Comment by Gamedev1909** [comment]
no i need the build with those changes drchat said i rather buy games

**Comment by Razzile** [comment]
it's good to see vulkan making progress for AMD.

@Gamedev1909 for the record marble blast was removed from xbox live years ago. the only way to get it is either from an xbox that had it before or from someone else who has it

**Comment by Gamedev1909** [comment]
i have it on my xbox 360 so im good

**Comment by SakataGintokiYT** [comment]
@DrChat 
https://github.com/RPCS3/rpcs3/blob/master/rpcs3/Emu/RSX/VK/VKGSRender.cpp#L328

**Comment by Gamedev1909** [comment]
ok im using codexl to debug and capture frames from sonic unleashed

**Comment by Gamedev1909** [comment]
from renderdoc  
[xen.zip](https://github.com/benvanik/xenia/files/775637/xen.zip)

[xenlog.zip](https://github.com/benvanik/xenia/files/775634/xenlog.zip)

**Comment by Gamedev1909** [comment]
vk_icdNegotiateLoaderICDInterfaceVersion - amdvlk64.dll
vk_icdNegotiateLoaderICDInterfaceVersion - amdvlk64.dll
vk_icdNegotiateLoaderICDInterfaceVersion - amdvlk64.dll
vk_icdNegotiateLoaderICDInterfaceVersion - amdvlk64.dll
vk_icdNegotiateLoaderICDInterfaceVersion - amdvlk64.dll
vk_icdNegotiateLoaderICDInterfaceVersion - amdvlk64.dll
0x00007ff7aa7f9fe6 - xenia.exe
0x00007ff7aa7f3281 - xenia.exe
0x00007ff7aa7dd259 - xenia.exe
0x00007ff7aa7dbcd6 - xenia.exe
0x00007ff7aa7db60c - xenia.exe
0x00007ff7aa7dbbde - xenia.exe
0x00007ff7aa7db41c - xenia.exe
0x00007ff7aa7dab6d - xenia.exe
0x00007ff7aa7ddcec - xenia.exe
0x00007ff7aa821788 - xenia.exe
0x00007ff7aa81fc7a - xenia.exe
0x00007ff7aa70513b - xenia.exe
BaseThreadInitThunk - KERNEL32.DLL


your call stack

**Comment by Gamedev1909** [comment]
your callstack for banjo will be soon

**Comment by Gamedev1909** [comment]
here is banjo k with the mods you asked us to do 

vk_icdNegotiateLoaderICDInterfaceVersion - amdvlk64.dll
vk_icdNegotiateLoaderICDInterfaceVersion - amdvlk64.dll
vk_icdNegotiateLoaderICDInterfaceVersion - amdvlk64.dll
vk_icdNegotiateLoaderICDInterfaceVersion - amdvlk64.dll
vk_icdNegotiateLoaderICDInterfaceVersion - amdvlk64.dll
vk_icdNegotiateLoaderICDInterfaceVersion - amdvlk64.dll
xe::gpu::vulkan::RenderCache::BeginRenderPass - render_cache.cc, line 688
xe::gpu::vulkan::VulkanCommandProcessor::IssueDraw - vulkan_command_processor.cc, line 567
xe::gpu::CommandProcessor::ExecutePacketType3_DRAW_INDX_2 - command_processor.cc, line 1117
xe::gpu::CommandProcessor::ExecutePacketType3 - command_processor.cc, line 591
xe::gpu::CommandProcessor::ExecuteIndirectBuffer - command_processor.cc, line 416
xe::gpu::CommandProcessor::ExecutePacketType3 - command_processor.cc, line 558
xe::gpu::CommandProcessor::ExecuteIndirectBuffer - command_processor.cc, line 416
xe::gpu::CommandProcessor::ExecutePacketType3 - command_processor.cc, line 558
xe::gpu::CommandProcessor::ExecutePrimaryBuffer - command_processor.cc, line 394
xe::gpu::CommandProcessor::WorkerThreadMain - command_processor.cc, line 161
std::_Func_impl<<lambda_d768b97bf2db2f2fe0223def01019a68>,std::allocator<int>,int>::_Do_call - functional, line 212
xe::kernel::XHostThread::Execute - xthread.cc, line 1048
<lambda_2d0f1f01f682aa03618b540ad143a4e0>::operator() - xthread.cc, line 384
xe::threading::ThreadStartRoutine - threading_win.cc, line 428
BaseThreadInitThunk - KERNEL32.DLL

**Comment by Gamedev1909** [comment]
hope that helps

**Comment by woody2014** [comment]
Any news for Amd people

**Comment by Gamedev1909** [comment]
so drchat any other ideas we can try ?

**Comment by SakataGintokiYT** [comment]
@Gamedev1909 
@DrChat 
Dragon Ball Raging Blast at AMD graphic card 
https://www.youtube.com/watch?v=YGB8LDB6HIo&t=0s

**Comment by Gamedev1909** [comment]
Thats odd every game i tested showed nothing new

**Comment by DrChat** [comment]
I've made a funding campaign for an AMD card [here](https://fundrazr.com/81DYF1).

I'll use the funds to possibly buy an [RX 480](https://www.amazon.com/STRIX-Radeon-Polaris-Graphics-STRIX-RX480-O8G-GAMING/dp/B01MU0OIN5) card, unless if anyone has any better recommendations.

**Comment by neolightning** [comment]
Get the RX480 if you want to use it for more than just xenia, if you just want one for xenia, pretty much any Vulkan compatible Radeon Card should work

**Comment by neolightning** [comment]
just go with whatever feels best for your needs

**Comment by Gamedev1909** [comment]
at least you can have tier 3 vulkan and no longer be restricted like with nvidia missing a ton of low level api hardware features

**Comment by DrChat** [comment]
![](http://i.imgur.com/GoLcLWa.png)

As posted on the fundrazr page, I've bought the card and it'll be here tomorrow.

**Comment by NuclearC** [comment]
we are gonna get Xenia on AMD soon :D

**Comment by Margen67** [comment]
>soon

™ :^)

**Comment by DrChat** [comment]
AMD people: You should start seeing stuff as of 855ebb9cfe0b89f707f0be5e1dcd25b8acb438a6.

![](http://i.imgur.com/Du1LlXL.jpg)

**Comment by shadowbrony33** [comment]
Nice!

**Comment by Gamedev1909** [comment]
ok cool im going to do some debugging on color errors and stuff and send it to you here

**Comment by Gamedev1909** [comment]
https://ci.appveyor.com/project/benvanik/xenia/build/artifacts best to post your appveyor on main page

**Comment by Gamedev1909** [comment]
drchat what are the settings for render doc or codexl it crashes on me

**Comment by Irixion** [comment]
On AMD (R9 270X) here--Not seeing too much. Just HUD and BG stuff. Seems like anything that's 3D isn't being rendered. On a side note, why aren't there any artifacts for recent builds?

**Comment by Gamedev1909** [comment]
Everything Crashes again

**Comment by Irixion** [comment]
![screenie](https://cloud.githubusercontent.com/assets/26639385/26033914/6a73231a-3882-11e7-9904-a661d92e1adb.PNG)

All of the titles I'm loading (that manage to get in game) don't render anything. Effects are showing up, but everything else isn't showing up.

**Comment by Gamedev1909** [comment]
newest build all games but star ocean 4 crash

**Comment by gibbed** [comment]
If you're going to report such issues, you need to be more specific than "all games but <game> crash".

**Comment by Irixion** [comment]
@gibbed 

I know that after I updated AMD's drivers to the latest one, Xenia would crash on Vulkan. I rolled back and am now fine. I'm not sure if that's related to the crashes Gamedev is experiencing.

**Comment by Gh0stBlade** [comment]
@Irixion Useful if you can provide the specific version of the driver and your GPU model.

**Comment by Irixion** [comment]
@Gh0stBlade Sure.

AMD R9 270x Driver version is 21.19.137.514 (Crimson 16.6) WHQL Win 10 64

**Comment by Irixion** [comment]
@Gh0stBlade Previous to build 44e7c37, I would get nothing, so that seems to have at least gotten the 2D elements to display.

**Comment by Gamedev1909** [comment]
clicks load selects iso then crash....
repeat for all games 
now if i did more detail like codexl or other tools he couldnt use it anyway so not much i can do

**Comment by Irixion** [comment]
@Gamedev1909 I would uninstall the driver and let Windows install it. AMD's drivers are buggy at best. Though you could always upload a log so we can see what's causing it.

**Comment by Gamedev1909** [comment]
no this driver isnt buggy and for streaming with avc this is the only one that works 16.6 is way to old has major issues with other emulators that i have

**Comment by Irixion** [comment]
...Let Windows install the optimal driver. Newer drivers have issues. Post the log so we can figure out why it's crashing (I don't feel like updating only to roll back)

**Comment by Gamedev1909** [comment]
no thanks i wont install a busted driver that prevents me from work that is counter productive not saying it again

**Comment by Irixion** [comment]
If Windows' default drivers aren't working, then you have some other issues with your system. 

The latest Crimson version is 17.4.4. That's unrelated to the driver version.   21.19.137.514 is 5 months old, not to mention I have a different card. Please post the log so we can have a look at it.

**Comment by Gamedev1909** [comment]
no newest crimson is 17.5.1 and it was released may 5th 2017 gl4 works as needed so until vulkan is updated or fixed i will avoid testing for now good luck guys

**Comment by Irixion** [comment]
15.5.1 is an optional update that only has bug fixes, and isn't the recommended driver by AMD. If anyone else presents the issue, I'll update it myself seeing as @Gamedev1909 isn't uploading the log.

OpenGL backend is deprecated, and isn't being maintained. Once Vulkan gets some TLC, I'm assuming GL will be removed.

**Comment by Irixion** [comment]
@DrChat @gibbed So I loaded a cut scene and this happened:
![screenie](https://cloud.githubusercontent.com/assets/26639385/26039896/f44d6f94-38ed-11e7-96ac-f6f1785e5e90.PNG)

Looks like some textures and part of the cut scene (flames) loading amongst the garbled mess. Here's the log for that:

[xenia.txt](https://github.com/benvanik/xenia/files/999909/xenia.txt)

For good measure this is on the 4260e381 commit. Hope this helps!

**Comment by Gamedev1909** [comment]
here are a few logs

[Star ocean 4.txt](https://github.com/benvanik/xenia/files/1008701/Star.ocean.4.txt)
[Banjo.txt](https://github.com/benvanik/xenia/files/1008704/Banjo.txt)
[Eternal-Sonata-NTSC-Xbox360.txt](https://github.com/benvanik/xenia/files/1008708/Eternal-Sonata-NTSC-Xbox360.txt)
[Halo SPartan Assult.txt](https://github.com/benvanik/xenia/files/1008711/Halo.SPartan.Assult.txt)
[minecraft.txt](https://github.com/benvanik/xenia/files/1008713/minecraft.txt)
[PerfectDark.txt](https://github.com/benvanik/xenia/files/1008714/PerfectDark.txt)
[RedDead.txt](https://github.com/benvanik/xenia/files/1008715/RedDead.txt)
[Sonic 2006.txt](https://github.com/benvanik/xenia/files/1008716/Sonic.2006.txt)
[Sonic unleashed.txt](https://github.com/benvanik/xenia/files/1008717/Sonic.unleashed.txt)
[Dark sector.txt](https://github.com/benvanik/xenia/files/1008719/Dark.sector.txt)

[Forza2.txt](https://github.com/benvanik/xenia/files/1008724/Forza2.txt)

**Comment by Gh0stBlade** [comment]
I can also confirm I'm also experiencing crashes after updating to the latest AMD drivers. 
vkCreateImage returns VK_ERROR_INITIALIZATION_FAILED

**Comment by Irixion** [comment]
@Gamedev1909 
@Gh0stBlade 

I will mention that AMD's Recommended Driver is 17.4.4, and not 17.5.1.

```
K> 0000011C XThread::Execute thid 26 (handle=0000011C, 'NoSyncThread (0000011C)', native=00001928)
i> 0000011C VdSwap(BFAA4C10, 4725FA70, BFFF3008, 4725FAB0, BEEF0001, 4725FA50(BF708000), 4725FA5C(00000006), 4725FA60(00000000), 4725FA58(00000500), 4725FA54(000002D0))
i> 0000011C VdSwap(BFAB3AEC, 4725FA70, BFFF3008, 4725FAB0, BEEF0001, 4725FA50(BF708000), 4725FA5C(00000006), 4725FA60(00000000), 4725FA58(00000500), 4725FA54(000002D0))
!> 00000004 Vulkan check: vkCreateImage returned VK_ERROR_INITIALIZATION_FAILED
```

That being said, the latest commit, 4e6a5147 gives me this error, regardless of driver version. My current driver, and the latest driver as well.

@DrChat Is there a specific version of Vulkan that we're supposed to have?

[xenia.txt](https://github.com/benvanik/xenia/files/1009032/xenia.txt)

**Comment by Gamedev1909** [comment]
As i said sooo many times im not switching drivers to solve a issue that shouldnt be going on with rpcs3 vulkan doesnt stop working and cause mass crashing because of a driver change and that shouldnt happen here either so i gave my logs

**Comment by DrChat** [comment]
At the moment AMD cards are crashing because they don't support D24
formats. I'm still working on a fix that isn't a complete hack.

**Comment by Irixion** [comment]
@Gamedev1909 You shouldn't be complaining. RPCS3 has a different team. Given the frequency of the updates there, I'm going to assume they have a larger team. The problem has been identified and fingers crossed, we'll see a solution soon. Thanks to all the hard work the contributes do! <3

**Comment by Gamedev1909** [comment]
it's not complaining it's informing you that i wont be switching drivers for the 6TH time so far don't ya think that 6 times is a little to much to try and convince a guy to go back to slower drivers... and no they dont have a larger team they just have more users more bug reports and they have @kd-11 who is a god among men for api fixes @DrChat maybe ask him for a fix he might have solved the issues already

**Comment by Irixion** [comment]
AMD cards don't support D24 formats? Lame. I'm sticking with team green from now on. Seeing as this issue is probably a mammoth to fix, fingers crossed. Unfortunately my experience isn't enough for me to become a contributor.

@Gamedev1909 Slower drivers? ...That's not how it works. I merely recommended that you install the same drivers that AMD recommends for 200 series cards.

**Comment by Gamedev1909** [comment]
Amd doesn't recommend anything and seeing how the games I play are faster on 17.5.1 that is my best option also same for my render work and 3D modelling again no thank you

**Comment by paulsapps** [comment]
RPCS3 has a bigger team, even said it yourself. More testers, more users, more devs. Turns out that makes things progress faster.

**Comment by Irixion** [comment]
@Gamedev1909 
![untitled](https://cloud.githubusercontent.com/assets/26639385/26232350/b10ac842-3c23-11e7-985e-4e70e2642c6a.png)

You were saying? Drivers don't "magically" make things "faster". They fix bugs that cause performance issues. AMD has release notes for every update. The problem with being on the "bleeding edge" is that there are always bugs left to sort out. The latest driver now is 17.5.2. That being said, if something works on one driver version, and not on another, you can assume that it is the driver causing it. AMD's drivers aren't that great anyway, and they seem to break two things for every one thing they end up fixing. The fact that their cards don't support D24 formats is the last straw for me, because clearly AMD is inferiour in a lot of areas.

Instead of arguing, you could just, you know, agree to disagree. You don't want to change your drivers? More power to you. I don't care.  Being on the bleeding edge updates can cause issues sometimes. I asked you to post logs before any changes to Xenia were made. Seeing as the recent commits of Xenia try to initialize D24 on AMD cards, regardless of driver, it doesn't really matter now. Not until the developers get some sort of workaround in place, which is probably going to take a long time seeing as it isn't something anyone can put together in a few hours.

Again, many thanks to all contributes to this project. You're the real MVPs. Out of curiosity though, is it all AMD cards that don't support this format? Is it on a hardware or driver level? All AMD cards I've had are more trouble than they are worth.

**Comment by Gamedev1909** [comment]
Let it go already you went from forcing advice to being a sad little person nothing your doing is helping keep the whining down to a min thanks

**Comment by DrChat** [comment]
No more arguing. This issue is locked until for now.

**Comment by DrChat** [comment]
This issue is no longer necessary.
If you have issues with Vulkan, create a new issue.
If you have questions, go to [the subreddit](https://www.reddit.com/r/popular/).
